packages/core/rendering/core-rendering-server-internal/src/render_utils.ts

```diff
@@ -25,14 +26,44 @@ export const getSettingValue = <T>(
 export const getBundlesHref = (baseHref: string, buildNr: string): string =>
   `${baseHref}/${buildNr}/bundles`;
 
-export const getStylesheetPaths = ({
-  themeVersion,
+export const getScriptPaths = ({
+  baseHref,
+  buildNum,
   darkMode,
+}: {
+  buildNum: number;
+  baseHref: string;
+  darkMode: DarkModeValue;
+}) => {
+  if (darkMode === 'system') {
+    return [`${baseHref}/ui/legacy_theme.js`];
+  } else {
+    return [];
+  }
+};
+
+export const getCommonStylesheetPaths = ({
+  baseHref,
+  buildNum,
+}: {
+  buildNum: number;
+  baseHref: string;
+}) => {
+  const bundlesHref = getBundlesHref(baseHref, String(buildNum));
+  return [
+    `${bundlesHref}/kbn-ui-shared-deps-src/${UiSharedDepsSrc.cssDistFilename}`,
+    `${baseHref}/ui/legacy_styles.css`,
+  ];
+};
+
+export const getThemeStylesheetPaths = ({
+  darkMode,
+  themeVersion,
```

Comment:
This is for point `2. Per-theme css files` from the issue's description

I split the "common" css files (not theme dependant - injected in the template) from the "theme" css files (theme dependant - no longer injected in the template - injected browser-side depending on the theme instead)

Reply:
nice